extensions/wikia/FilePage/WikiaFilePage.php

```diff
@@ -148,4 +148,87 @@ protected function newPage( Title $title ) {
 		return new WikiaWikiFilePage( $title );
 	}
 
+	/**
+	 * Render the image or video
+	 */
+	public function view() {
+		global $wgMemc;
+		$out = $this->getContext()->getOutput();
+		if ( !$this->getContext()->getUser()->isAnon() ) {
+			parent::view();
+
+			return;
+		}
+		//fallback to main page
+		$url = Title::newMainPage()->getFullURL();
+		//wiki needs read privileges
+		if ( !$this->getTitle()->userCan( 'read' ) ) {
+			$out->redirect( $url , '301');
+			return;
+		}
+		$redirKey = wfMemcKey( 'redir', $this->getTitle()->getPrefixedText() );
+
+		$displayImg = $img = false;
+		Hooks::run( 'ImagePageFindFile', [ $this, &$img, &$displayImg ] );
+		if ( !$img ) { // not set by hook?
+			$img = wfFindFile( $this->getTitle() );
+			if ( !$img ) {
+				$img = wfLocalFile( $this->getTitle() );
+			}
+		}
+
+		if ( !$img || $img && !$img->fileExists ) {
+			$wgMemc->add( $redirKey, $url );
+			$out->redirect( $url , '301' );
+			return;
+		}
+
+		$urlMem = $wgMemc->get( $redirKey );
+		if ( $urlMem ) {
+			$out->redirect( $urlMem , '301' );
+
+			return;
+		}
+
+		$res = $this->fetchLinks( $img->getTitle()->getDBkey() );
+
+		foreach ( $res as $row ) {
+			$title = Title::newFromRow( $row );
+			if ( $title->isRedirect() ) {
+				continue;
+			}
+			if ( !$title->userCan( 'read' ) ) {
+				$out->redirect( $url , '301');
+				continue;
+			}
+			$url = $title->getFullURL();
+			$res->free();
```

Comment:
You do not free the resource when one of the above if statements is true. Is that intentional?

Reply:
I think it can be deleted. Right now it isn't needed.